cms/admin/placeholderadmin.py

```diff
@@ -419,59 +447,128 @@ def edit_plugin(self, request, plugin_id):
+        try:
+            plugin_id = get_int(request.POST.get('plugin_id'))
+        except TypeError:
+            raise RuntimeError("'plugin_id' is a required parameter.")
+        plugin = CMSPlugin.objects.get(pk=plugin_id)
+        try:
+            placeholder_id = get_int(request.POST.get('placeholder_id'))
+        except TypeError:
+            raise RuntimeError("'placeholder_id' is a required parameter.")
+        except ValueError:
+            raise RuntimeError("'placeholder_id' must be an integer string.")
+        placeholder = Placeholder.objects.get(pk=placeholder_id)
+        # The rest are optional
+        parent_id = get_int(request.POST.get('plugin_parent', ""), None)
         language = request.POST.get('plugin_language', None)
+        move_a_copy = request.POST.get('move_a_copy', False)
+        move_a_copy = (move_a_copy and move_a_copy != "0" and
+                       move_a_copy.lower() != "false")
+
         source_placeholder = plugin.placeholder
-        if not parent_id:
-            parent_id = None
-        else:
-            parent_id = int(parent_id)
         if not language and plugin.language:
             language = plugin.language
         order = request.POST.getlist("plugin_order[]")
+
         if not self.has_move_plugin_permission(request, plugin, placeholder):
-            return HttpResponseForbidden(force_text(_("You have no permission to move this plugin")))
-        if not placeholder == source_placeholder:
+            return HttpResponseForbidden(
+                force_text(_("You have no permission to move this plugin")))
+        if placeholder != source_placeholder:
             try:
                 template = self.get_placeholder_template(request, placeholder)
-                has_reached_plugin_limit(placeholder, plugin.plugin_type, plugin.language, template=template)
+                has_reached_plugin_limit(placeholder, plugin.plugin_type,
+                                         plugin.language, template=template)
             except PluginLimitReached as er:
                 return HttpResponseBadRequest(er)
-        if parent_id:
-            if plugin.parent_id != parent_id:
+
+        if move_a_copy:  # "paste"
+            if plugin.plugin_type == "PlaceholderPlugin":
+                inst, _plugin = plugin.get_plugin_instance()
+                new_plugins = copy_plugins.copy_plugins_to(
+                    inst.placeholder_ref.get_plugins(), placeholder,
+                    to_language=language)
+            else:
+                source_plugins = [plugin] + list(plugin.get_descendants())
+                new_plugins = copy_plugins.copy_plugins_to(
+                    source_plugins, placeholder, language, parent_id)
+
+            new_plugins = [nu for nu, _old in new_plugins]
+            top_parent = new_plugins[0].parent_id
+            top_plugins = [p for p in new_plugins if p.parent_id == top_parent]
+            top_plugins_pks = [str(p.pk) for p in top_plugins]
```

Comment:
maybe it would be better to combine ``top_plugins`` and ``top_plugins_pks`` (and also maybe ``new_plugins ``) into one for loop instead of 2 (3) list comprehensions?

Though minor importance.